integration-libs/opf/components/opf-checkout-payment-wrapper/opf-checkout-payment-wrapper.service.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { ActiveCartService } from '@spartacus/cart/base/core';
+import { RoutingService, UserIdService } from '@spartacus/core';
+import { OpfResourceLoaderService } from '@spartacus/opf/core';
+import {
+  OpfCheckoutFacade,
+  OpfOtpFacade,
+  OpfPaymentMethodType,
+  OpfRenderPaymentMethodEvent,
+  PaymentSessionData,
+} from '@spartacus/opf/root';
+
+import { BehaviorSubject, Observable, combineLatest } from 'rxjs';
+import { filter, map, switchMap } from 'rxjs/operators';
+
+@Injectable()
+export class OpfCheckoutPaymentWrapperService {
+  constructor(
+    protected opfCheckoutService: OpfCheckoutFacade,
+    protected opfOtpService: OpfOtpFacade,
+    protected opfResourceLoaderService: OpfResourceLoaderService,
+    protected userIdService: UserIdService,
+    protected activeCartService: ActiveCartService,
+    protected routingService: RoutingService
+  ) {}
+
+  protected activeCartId: string;
+
+  protected renderPaymentMethodEvent$ =
+    new BehaviorSubject<OpfRenderPaymentMethodEvent>({
+      isLoading: false,
+    });
+
+  getRenderPaymentMethodEvent(): BehaviorSubject<OpfRenderPaymentMethodEvent> {
+    return this.renderPaymentMethodEvent$;
+  }
+
+  initiatePayment(paymentOptionId: number): Observable<PaymentSessionData> {
+    this.renderPaymentMethodEvent$.next({
+      isLoading: true,
+    });
+
+    return combineLatest([
+      this.userIdService.getUserId(),
+      this.activeCartService.getActiveCartId(),
+    ]).pipe(
+      switchMap(([userId, cartId]) => {
+        this.activeCartId = cartId;
+        return this.opfOtpService.generateOtpKey(userId, cartId);
+      }),
+      filter((response) => Boolean(response?.value)),
+      map(({ value: otpKey }) => {
```

Comment:
I think we could move part from `map` to separate method in the service, to be closer to single responsibility and make this method leaner. To have something like: `map(({ value: otpKey }) => this.someName(paymentOptionId))`